### PR TITLE
Adjust RecomputeGeneral class ...

### PR DESCRIPTION
**Pull request by jasondhales**
Adjust Recompute classes.  This will enable use of temperature-dependent moduli.  No tests were changed as this change enables future development but does not alter end behavior.  #5481

### CODE REVIEW

File: modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C [hunk 1]
```diff
@@ -38,8 +37,8 @@ ComputeReturnMappingStress::computeQpStress()
   // compute the stress increment (as a tensor) and the inelastic strain increment (tensor) required to return to the yield surface
```
Comment (sapitts) [inline modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C]: Would you mind updating this comment to clarify that the whole stress, not the stress increment, is returned please?

Comment (jasondhales) [inline modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C]: Done.

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C [hunk 2]
```diff
@@ -6,10 +6,12 @@
 /****************************************************************/
 #include "RecomputeRadialReturnIsotropicPlasticity.h"
 
+#include "Function.h"
```
Comment (jwpeterson) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C]: Did this include need to be added?  I didn't see any other changes in the file that would appear to require it.

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C [hunk 3]
```diff
@@ -19,24 +19,24 @@ InputParameters validParams<RecomputeRadialReturnStressIncrement>()
...
 
-RecomputeRadialReturnStressIncrement::RecomputeRadialReturnStressIncrement(const InputParameters & parameters) :
-    RecomputeGeneralReturnStressIncrement(parameters),
-    _output_iteration_info(getParam<bool>("output_iteration_info")),
-    _output_iteration_info_on_error(getParam<bool>("output_iteration_info_on_error")),
-    _relative_tolerance(parameters.get<Real>("relative_tolerance")),
-    _absolute_tolerance(parameters.get<Real>("absolute_tolerance"))
+RecomputeRadialReturn::RecomputeRadialReturn(const InputParameters & parameters) :
+  RecomputeGeneralReturn(parameters),
+  _output_iteration_info(getParam<bool>("output_iteration_info")),
```
Comment (jwpeterson) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C]: I think this indent needs to stay at 4 spaces, or @andrsd gets mad.

File: modules/tensor_mechanics/src/materials/RecomputeGeneralReturn.C [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  params.addParam<unsigned int>("max_iterations", 30, "Maximum number of newton iterations in the radial return material");
+  params.addParam<std::string>("base_name", "Optional parameter that allows the user to define multiple mechanics material systems on the same block, i.e. for multiple phases");
+  params.set<bool>("compute") = false; //The return stress increment classes are intended to be iterative materials
+  return params;
+}
+
+
+RecomputeGeneralReturn::RecomputeGeneralReturn(const InputParameters & parameters) :
+  Material(parameters),
+  _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "" ),
```
Comment (jwpeterson) [inline modules/tensor_mechanics/src/materials/RecomputeGeneralReturn.C]: Also 4 spaces here.

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C (hunk 3 above)
Comment (jasondhales) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C]: OK.  I'll change it to four spaces.  We all realize this is silly, right?  Why are four spaces in an initialization list better than two (used everywhere else)?

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C (hunk 2 above)
Comment (jasondhales) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C]: Won't compile without it.

**Comment by permcody** [comment]
👍

**Pull request merged**